### PR TITLE
Log actions in Web UI (v2)

### PR DESCRIPTION
**Pull request by patrickleotardif**
Involves
- creating separate endpoints for previously combined "actions"
- tweaking Log model
- using a function decorator as suggested for lightweight logging

@mistercrunch

### CODE REVIEW

**Comment by patrickleotardif** [comment]
@mistercrunch all unit tests green. Take a quick look?

File: airflow/www/utils.py [hunk 1]
```diff
@@ -13,6 +13,9 @@
 from wtforms.compat import text_type
 
 from airflow.configuration import conf
+from airflow import login
```
Comment (mistercrunch) [inline airflow/www/utils.py]: Let's bunch those:
`from airflow import login, models, settings`

File: airflow/www/utils.py [hunk 2]
```diff
@@ -68,6 +71,33 @@ def limit_sql(sql, limit, conn_type):
...
+                user = login.current_user.username
+            else:
+                user = 'anonymous'
+
+            session.add(
+                models.Log(
+                    event=action,
+                    task_instance=None,
+                    owner=user
+                    ))
```
Comment (mistercrunch) [inline airflow/www/utils.py]: nit: i'd bring `))` to the end of line 93

File: airflow/www/views.py [hunk 3]
```diff
@@ -834,10 +834,40 @@ def task(self):
             form=form,
             dag=dag, title=title)
 
-    @expose('/action')
+    @expose('/run')
     @login_required
-    def action(self):
-        action = request.args.get('action')
+    @wwwutils.action_logging('run')
```
Comment (mistercrunch) [inline airflow/www/views.py]: nice!

File: airflow/www/utils.py [hunk 4]
```diff
@@ -68,6 +71,33 @@ def limit_sql(sql, limit, conn_type):
...
+            if hasattr(login.current_user, 'username'):
+                user = login.current_user.username
+            else:
+                user = 'anonymous'
+
+            session.add(
+                models.Log(
+                    event=action,
+                    task_instance=None,
+                    owner=user
```
Comment (mistercrunch) [inline airflow/www/utils.py]: it'd be really nice to also store a string representation of args and kwargs as well. Maybe we can add an `extra` blob to the Log model and store a json representation of both these things as in `{"args": ['foo'], "kwargs": {"key": "bar"}}`

We'll need a db migration script to add the field. It's easy but unintuitive at first. I can guide you through the process.

**Comment by mistercrunch** [comment]
I like the refactor on the endpoints, this is nice! We should probably log all events now that it's so easy.

**Comment by patrickleotardif** [comment]
all green and tested on sqlite. will let you take final look @mistercrunch

**Comment by mistercrunch** [comment]
img

**Pull request merged**